CIViC/load_stage.sql

```diff
@@ -0,0 +1,109 @@
+/**************************************************************************
+* Copyright 2016 Observational Health Data Sciences and Informatics (OHDSI)
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* Authors: Varvara Savitskaya, Vlad Korsik
+* Date: 2022
+**************************************************************************/
+
+DO $_$
+BEGIN
+	PERFORM VOCABULARY_PACK.SetLatestUpdate(
+	pVocabularyName			=> 'CIViC',
+	pVocabularyDate			=> current_date,
+	pVocabularyVersion		=> 'CIViC'||TO_CHAR(CURRENT_DATE,'YYYYMMDD'),
+	pVocabularyDevSchema	=> 'dev_civic'
+);
+END $_$;
+
+
+truncate concept_stage;
+truncate concept_relationship_stage;
+truncate concept_synonym_stage;
+
+drop table civic_source;
+create table civic_source as (
+select distinct variant as concept_name, 'CIViC' as vocabulary_id, cast(variant_id as varchar(5)) as concept_code, ( regexp_matches(hgvs_expressions, '[^, ]+', 'g'))[1] as hgvs
+from sources.civic_variantsummaries
+where hgvs_expressions ~ '[\w_]+(\.\d+)?:[cCgGoOmMnNrRpP]\.'
+and variant!~*'frameshift|truncating'
+
+union
+
+select distinct variant as concept_name, 'CIViC' as vocabulary_id, cast(variant_id as varchar(5)) as concept_code, concat(gene, ':p.', variant) as hgvs
+from sources.civic_variantsummaries
+where variant ~'([A-Z][1-9]*[A-Z])'
+and variant!~*'expression|amplification|truncation|truncating|loss|wildtype|mutation|methylation|polymorphism|HOMOZYGOSITY|translocation|PHOSPHORYLATION|deletion|function|shift|alteration|tandem|serum|alternative|REARRANGEMENT|MISLOCALIZATION|and|INACTIVATION|DOMAIN'
+and variant!~*'rs'
+
+union
+
+select distinct variant as concept_name, 'CIViC' as vocabulary_id, cast(variant_id as varchar(5)) as concept_code, variant as hgvs
+from sources.civic_variantsummaries
+where variant ~'([A-Z][1-9]*[A-Z])'
+and variant!~*'expression|amplification|truncation|truncating|wildtype|mutation|loss|methylation|polymorphism|HOMOZYGOSITY|translocation|PHOSPHORYLATION|deletion|function|shift|alteration|tandem|serum|alternative|REARRANGEMENT|MISLOCALIZATION|and|INACTIVATION|DOMAIN'
+and variant~*'rs'
+and variant!~'::');
+
+
+--insert into concept_stage
+insert into concept_stage
+with s as (SELECT DISTINCT
+       COALESCE(c.concept_id, NULL::int)  as concept_id,
+       trim(substr(n.concept_name,1,255)) AS concept_name,
```

Comment:
don't do that, use vocabulary_pack.CutConceptName() (see our "SQL technical guide, common problems and tricks", p.1)